src/main/java/org/optaplanner/springboottaskassigning/TaskAssigningSolverManagerController.java

```diff
@@ -34,50 +30,46 @@
 import org.springframework.web.server.ResponseStatusException;
 
 @RestController
-@RequestMapping("/tenants/{tenantId}/solver")
+@RequestMapping("/tenants/{problemId}/solver")
 public class TaskAssigningSolverManagerController {
 
-    private final Logger logger = LoggerFactory.getLogger(this.getClass());
-
     @Autowired
     private TaskAssigningSolverManagerService solverManagerService;
 
     @PostMapping
-    public void solve(@PathVariable Long tenantId, @RequestBody TaskAssigningSolution planningProblem) {
-        logger.debug("POST /tenants/{}/solver", tenantId);
-        solverManagerService.solve(tenantId, planningProblem);
+    public void solve(@PathVariable Long problemId, @RequestBody TaskAssigningSolution planningProblem) {
+        if (!solverManagerService.solve(problemId, planningProblem)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Problem ( " + problemId + ") already exists.");
```

Comment:
Could you please also add some negative tests for cases like this one? To make sure the client gets answer we expect.

Also as I glanced over the TaskAssigningSolverManagerControllerTest, I have some suggestions to think about:
- using randoms in tests is generally risky, as you want the test to be deterministic
- there are some repeated REST calls, which could be put into private methods

Reply:
I have a `FIXME` annotation for the random usage, I'll either find a way to set a seed or remove it altogether in a later PR.